src/main/java/lab5/reflections.md

```diff
@@ -0,0 +1,99 @@
+Ques.1   Why a HumanPlayer can be returned when the return type is Player. ?
+Ans....  This is possible because of polymorphism in Java, specifically subtype polymorphism (also known as dynamic method dispatch).
+        - In Java, a subclass object can be assigned to a superclass reference.
+        - The promptForPlayer method has a return type of Player, which is an abstract class.
+        - HumanPlayer, Linus, and Omola are subclasses of Player, meaning they inherit from Player.
+        - Because Java supports upcasting (assigning a subclass object to a superclass reference), we can return instances of HumanPlayer, Linus, or Omola without changing the return type.
+
+
+Ques.2   Why pickNextMove must be an abstract method in Player and not just in Player subclasses. ?
+
+Ans.... Why Did the Error Occur Initially?
+  -->   The error occurred because the pickNextMove method was removed from the Player class and placed into the HumanPlayer subclass.
+        The variable whoseTurn is of type Player (the abstract superclass).
+        This caused a compilation error because Java did not know if whoseTurn (a Player) actually had a pickNextMove() method.
+
+Ans.... Why Does Adding the Abstract Method Fix the Error?
+   -->  By adding an abstract method signature for pickNextMove back into the Player class.
+        Java knows that every object of type Player (whether it's HumanPlayer, Linus, or Omola) must have a pickNextMove method.
+
+
+Ques.3   Why it is not necessary to change Main or TicTacToeGame in order to add new Player types to the game. ?
+Ans....  1). Polymorphism: The Key to Flexibility
+          -  Polymorphism allows us to treat different types of players (HumanPlayer, Linus, Omola, etc.) uniformly by referring to them through their shared superclass (Player).
+          -  Even though whoseTurn is a Player reference, at runtime, Java determines which specific implementation of pickNextMove() to call based on the actual object type (whether it's HumanPlayer, Linus, or Omola).
```

Comment:
This runtime determination is the specific behaviour that the term 'dynamic method dispatch' refers to, and you should mention it specifically